src/meta_schedule/database/memory_database.cc

```diff
@@ -71,7 +71,7 @@ class MemoryDatabaseNode : public DatabaseNode {
       if (run_secs.empty()) {
         continue;
       }
-      if (record->workload.same_as(workload)) {
+      if (WorkloadEqual(GetModuleEquality())(record->workload, workload)) {
```

Comment:
Would it be practical to have a common suite of `Database` tests that can be run against each implementation to test common invariants?

Reply:
Hi Peter, we do have unit tests for MetaSchedule Database [here](https://github.com/apache/tvm/blob/23c2909f2917fe9fbccad8e440199e5e4a106ba9/tests/python/unittest/test_meta_schedule_database.py) but there's not many cases against MemoryDatabase. Will add a regression test in this PR.